packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,688 @@
+  /**
+   * Import an existing LaunchTemplate.
+   */
+  public static fromLaunchTemplateAttributes(scope: Construct, id: string, attrs: LaunchTemplateAttributes): ILaunchTemplate {
+    const haveId = Boolean(attrs.launchTemplateId);
+    const haveName = Boolean(attrs.launchTemplateName);
+    if (haveId == haveName) {
+      throw new Error('LaunchTemplate.fromLaunchTemplateAttributes() requires exactly one of launchTemplateId or launchTemplateName be provided.');
+    }
+
+    class Import extends Resource implements ILaunchTemplate {
+      public readonly versionNumber = attrs.versionNumber ?? LaunchTemplateSpecialVersions.DEFAULT_VERSION;
+      public readonly launchTemplateId? = attrs.launchTemplateId;
+      public readonly launchTemplateName? = attrs.launchTemplateName;
+    }
+    return new Import(scope, id);
+  }
+
+  // ============================================
+  //   Members for ILaunchTemplate interface
+  /**
+   * @inheritdoc
+   */
+  public readonly versionNumber: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateId?: string;
+
+  /**
+   * @inheritdoc
+   */
+  public readonly launchTemplateName?: string;
+
+  // =============================================
+  //   Data members
+
+  /**
+   * The default version for the launch template.
+   *
+   * @attribute
+   */
+  public readonly defaultVersionNumber: string;
+
+  /**
+   * The latest version of the launch template.
+   *
+   * @attribute
+   */
+  public readonly latestVersionNumber: string;
+
+  /**
+   * IAM Role assumed by instances that are launched from this template.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Principal to grant permissions to.
+   */
+  public readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * Allows specifying security group connections for the instance.
+   *
+   * Note: If the LaunchTemplate was not created with a SecurityGroup, then this connections
+   * object will be empty and operations on it will do nothing.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * UserData executed by instances that are launched from this template.
+   */
+  public readonly userData?: UserData;
+
+
+  protected tagSpecifications: LaunchTemplateTagSpecification[];
+
+  // =============================================
+
+  constructor(scope: Construct, id: string, props?: LaunchTemplateProps) {
+    super(scope, id);
+
+    // Basic validation of the provided spot block duration
+    const spotDuration = props?.spotOptions?.blockDuration?.toHours({ integral: true });
+    if (spotDuration !== undefined && (spotDuration < 1 || spotDuration > 6)) {
+      // See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances
+      Annotations.of(this).addError('Spot block duration must be exactly 1, 2, 3, 4, 5, or 6 hours.');
+    }
+
+    this.role = props?.role ?? new iam.Role(this, 'Role', {
```

Comment:
Hrm. I'm thinking that it might not have been a good idea to default-create a Role for the launch template. Unfortunately, not doing so makes implementing `IGrantable` impossible.

Reason for not creating it would be a potential use-case for LaunchTemplate within the [LaunchTemplate spec](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-launchtemplate.html) of an ASG's [MixedInstancesPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-autoscaling-autoscalinggroup-mixedinstancespolicy.html) whereby the user provides a base LaunchTemplate that defines all of the common stuff (Role, UserData, AMI, etc) but then selectively overrides some of the non-Role fields (ex: AMI, CpuOptions, Placement) for certain instance types. If we default-create a Role, always, in the L2 then that sort of use-case ends up with the selective-override templates overriding the instance Role. 

On the other hand, a user could simply pass the role from the base LaunchTemplate into the override LaunchTemplates in that case. Sure, the override templates would be defining a Role but it would be the same Role as in the base template -- so, functionally equivalent.

Thoughts? Opinions?

Reply:
On the surface, it seems that it might be possible to use a [`LazyRole`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-iam.LazyRole.html)  here, and use a `Lazy` to create the `CfnInstanceProfile` only if the `LazyRole` has a statement, policy, or managed policy attached. Or if the `LazyRole` has been instantiated, I suppose. 

That's, perhaps, a little sketchy though.

Reply:
Yeah let's keep it as simple as we can for now, until we get a better grip on this.

Let's let `LaunchTemplates` do as little as possible, except just hold on to some common objects/variables.

Any method that would normally interact with those unset members can throw an exception for all I care. Let's see where that takes us. It's easier to add magic in than to take it out later.